### DISCUSSION TITLE
RequestValidationError: 'field required' error although no query param specified

### DISCUSSION THREAD

**Discussion by afrischk** [fastapi/fastapi#8427] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import uvicorn
from fastapi import FastAPI, Depends
from typing import Protocol

app = FastAPI()

class P(Protocol):
    def hello(self):
        ...

class Concrete(P):
    def hello(self):
        return "Hello World!"


class Bridge():
    def __init__(self, protocol: P = Depends(Concrete)):
        self.protocol = protocol

    def hello(self):
        return self.protocol.hello()


@app.get('/test')
def get(dep=Depends(Bridge)):
    return dep.hello()

if __name__ == "__main__":
    uvicorn.run("issue:app", host="0.0.0.0", port=5500, workers=1)
```

### Description

The example code throws a request validation error resulting in `422 Unprocessable Entity` with the following content:
```json
{
    "detail": [
        {
            "loc": [
                "query",
                "args"
            ],
            "msg": "field required",
            "type": "value_error.missing"
        },
        {
            "loc": [
                "query",
                "kwargs"
            ],
            "msg": "field required",
            "type": "value_error.missing"
        }
    ]
}
```
Inheritance from the Protocol triggers the error. If I remove the Protocol as a base class, it is working again as expected.

### Operating System

Linux

### FastAPI Version

0.88.0

### Python Version

3.10.0

### Additional Context

I am using pipenv.

**Comment by afrischk** [comment]
I have updated the issue with a minimal example to reproduce the error. Using `Depends` with a class that has a `Protocol` as a base class causes the error. Is there any workaround known?

**Comment by NewSouthMjos** [comment, marked as answer]
Yes, there is:

```
import uvicorn
from fastapi import FastAPI, Depends
from typing import Protocol

app = FastAPI()

class P(Protocol):
    # Error occurs in cause of this:
    # def __init__(self, *args, **kwargs) -> None:
    #     super().__init__(*args, **kwargs)

    # You can overwrite __init__ method:
    def __init__(self) -> None:
        super().__init__()

    def hello(self):
        ...

class Concrete(P):
    def hello(self):
        return "Hello World!"


class Bridge():
    def __init__(self, protocol: P = Depends(Concrete)):
        self.protocol = protocol

    def hello(self):
        return self.protocol.hello()


@app.get('/test')
def get(dep=Depends(Bridge)):
    return dep.hello()

if __name__ == "__main__":
    uvicorn.run("main:app", host="0.0.0.0", port=5600, workers=1)
```

/test works now as expected.

**Comment by afrischk** [comment]
Thanks! Works when I overwrite the `__init__()` function in the `Concrete`.
